packages/network/src/logic/node/ProxyStreamConnectionManager.ts

```diff
@@ -0,0 +1,211 @@
+                return
+            } else if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, targetNodeId)) {
+                const reason = `Could not open a proxy outgoing stream connection ${spid.key}, proxy stream connection already exists`
+                logger.warn(reason)
+                this.node.emit(Event.PUBLISH_STREAM_REJECTED, targetNodeId, spid, reason)
+                return
+            } else if (this.streamManager.isSetUp(spid) && this.hasConnection(targetNodeId, spid)) {
+                const reason = `Could not open a proxy outgoing stream connection ${spid.key}, a connection already exists`
+                logger.warn(reason)
+                return
+            }
+            this.addConnection(spid, targetNodeId)
+            await this.trackerManager.connectToSignallingOnlyTracker(trackerId, trackerAddress)
+            await promiseTimeout(this.nodeConnectTimeout, this.nodeToNode.connectToNode(targetNodeId, trackerId, false))
+            await this.nodeToNode.requestPublishOnlyStreamConnection(targetNodeId, spid)
+        } catch (err) {
+            logger.warn(`Failed to create a proxy outgoing stream connection to ${targetNodeId} for stream ${spid.key}:\n${err}`)
+            this.removeConnection(spid, targetNodeId)
+            this.node.emit(Event.PUBLISH_STREAM_REJECTED, targetNodeId, spid, err)
+        } finally {
+            this.trackerManager.disconnectFromSignallingOnlyTracker(trackerId)
+        }
+    }
+
+    async closeOutgoingStreamConnection(spid: SPID, targetNodeId: NodeId): Promise<void> {
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, targetNodeId)) {
+            clearTimeout(this.getConnection(targetNodeId, spid)!.reconnectionTimer!)
+            await this.nodeToNode.leaveStreamOnNode(targetNodeId, spid)
+            this.removeConnection(spid, targetNodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, targetNodeId, spid)
+        } else {
+            logger.warn(`A proxy outgoing stream connection for ${spid.key} on node ${targetNodeId} does not exist`)
+        }
+    }
+
+    processLeaveRequest(message: UnsubscribeRequest, nodeId: NodeId): void {
+        const { streamId, streamPartition } = message
+        const spid = new SPID(streamId, streamPartition)
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasInOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+        }
+        if (this.streamManager.isSetUp(spid) && this.streamManager.hasOutOnlyConnection(spid, nodeId)) {
+            this.removeConnection(spid, nodeId)
+            this.node.emit(Event.ONE_WAY_CONNECTION_CLOSED, nodeId, spid)
+            logger.info(`Proxy node ${nodeId} closed one-way stream connection for ${spid}`)
+        }
```

Comment:
This will not be executed in practice currently, so maybe its not required. It is there so that the proxy contact nodes would be able to remove proxied nodes if necessary